astropy/io/ascii/html.py

```diff
@@ -83,7 +83,14 @@ def process_lines(self, lines):
                                         'installed to read HTML tables')
 
         if 'parser' not in self.html:
-            soup = BeautifulSoup('\n'.join(lines))
+            try:
+                import warnings
+            except ImportError:
+                soup = BeautifulSoup('\n'.join(lines))
+            else:
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
```

Comment:
Catch *only* the warning in question, not *every* warning.  You might need to use `filterwarnings` instead of `simplefilter`, but you can figure that out.